### PR TITLE
adds assemblyai data reader

### PR DESCRIPTION
**Pull request by taigrr**
Adds support for AssemblyAI as a Reader

Submitted as part of the Verba Challenge for AI Tools Hack Night

### CODE REVIEW

**Comment by patrickloeber** [comment]
nice!!!! @thomashacker let me know if I can help with the review

File: goldenverba/components/reader/AssemblyAIAPI.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            config,
+            "API Key",
+            "ASSEMBLYAI_API_KEY",
+            "No AssemblyAI API Key detected",
+        )
+        aai.settings.api_key = token
+
+        # Validate quality
+        quality = config["Quality"].value
+        if quality not in ["nano", "best"]:
```
Comment (patrickloeber) [inline goldenverba/components/reader/AssemblyAIAPI.py:102]: we use "best" as default if no `speech_model` parameter is passed, so you could still allow it even if the user does not provide a quality parameter

File: goldenverba/components/reader/AssemblyAIAPI.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        file_data.add_field(
+            "files",
+            file_bytes,
+            filename=f"{fileConfig.filename}.{fileConfig.extension}",
+        )
+
+        try:
+            transcriber = aai.Transcriber(config=aaiConfig)
+            transcript = transcriber.transcribe(file_bytes)
+            if transcript.text is None:
```
Comment (patrickloeber) [inline goldenverba/components/reader/AssemblyAIAPI.py:122]:
```suggestion
            if transcript.error
```

File: goldenverba/components/reader/AssemblyAIAPI.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                raise Exception(
+                    f"AssemblyAI API failed to transcribe {fileConfig.filename}"
```
Comment (patrickloeber) [inline goldenverba/components/reader/AssemblyAIAPI.py:124]:
```suggestion
                    f"AssemblyAI API failed to transcribe {fileConfig.filename}: {transcript.error}"
```

File: goldenverba/components/reader/AssemblyAIAPI.py (hunk 2 above)
Comment (taigrr) [reply goldenverba/components/reader/AssemblyAIAPI.py:122]: Sounds good, I'll do both as an or clause

File: goldenverba/components/reader/AssemblyAIAPI.py (hunk 1 above)
Comment (taigrr) [reply goldenverba/components/reader/AssemblyAIAPI.py:102]: The input here is taken from the web dashboard, so it should be impossible to have a value other than these two anyway. But I can swap the default from bank to best.

File: goldenverba/components/reader/AssemblyAIAPI.py (hunk 3 above)
Comment (taigrr) [reply goldenverba/components/reader/AssemblyAIAPI.py:124]: Great

**Comment by taigrr** [comment]
Cool, I'll address these comments tomorrow morning

File: goldenverba/components/reader/AssemblyAIAPI.py (hunk 1 above)
Comment (patrickloeber) [reply goldenverba/components/reader/AssemblyAIAPI.py:102]: ah I see! Yeah I'd recommend making "best" the default 🙏

File: goldenverba/components/reader/AssemblyAIAPI.py (hunk 2 above)
Comment (patrickloeber) [reply goldenverba/components/reader/AssemblyAIAPI.py:122]: the text is always None if there is a `transcript.error`. But yeah being more explicit with both is fine

Comment (taigrr) [reply goldenverba/components/reader/AssemblyAIAPI.py:122]: But do I always get an error back if the text is empty? Like if I upload an mp3 of static noise or silence?

**Comment by thomashacker** [comment]
Awesome work everyone! ❤️ Let me know once you finished the changes, and then I'll have a quick last look

**Comment by taigrr** [comment]
CLA is A-Ok!

**Comment by taigrr** [comment]
thanks @thomashacker , please review

**Review by thomashacker — APPROVED** [review]
Looks great to me! Amazing work

**Pull request merged**